internal/declarative/v2/reconciler.go

```diff
@@ -81,7 +83,7 @@ func newResourcesCondition(obj Object) metav1.Condition {
 	}
 }
 
-//nolint:funlen,cyclop
+//nolint:funlen,cyclop, gocognit
```

Comment:
Any chance we can fix the issue reported by this new linter or would the changeset be too big?

Reply:
This `gocognit` basically means in this reconciliation loop, we have too much if else conditions and make the reconciliation flow complicated. But it can be easily reduced by extracting some logic into meaningful functions.

Just want to mention, with remove declarative library task, we will not change current reconciliation logic too much, maybe just rename some existing functions to more expressive compare to current genearic function.

For software refactoring, when we meet this kind of situation that linter is complain the complexities, it's always good to not ignore it and refactoring later, and in practice, in most cases, it will be neglected later.

I would say it worth take efforts to think avoid this `gocognit` complain, one hint, if you look at this `renderer`, it's actually only used in `pruneDiff`, and it can be encapsulated into that function, then you can reduce one if condition, and if you look further, this `renderer` only used in `renderer.RemovePrerequisites(ctx, obj)`, and currently, there is no need to remove any prerequisites, the implementation of `RemovePrerequisites` returns `nil`
```
renderer, err := InitializeRenderer(ctx, obj, spec, r.Options)
```
What I want to say is, take the newly introduced linter complain as a chance to revisit the code structure, it will make you start to think and even observe some tech debts in earlier stage.

But of course, regarding how to work with those tech debts, it depends, in most cases, it may not relavent to the current issue you working with, but then, we can create a follow-up issue, and plan to resolve later.

Reply:
I removed the renderer fully, since it has a nil implementation and the corresponding interface has only this one implementation.

Furthermore, removed the RemoveCRDs option, since this was always set to false, and thus I could also remove one more condition to simplify the code